apps/modernization-ui/src/filters/entry/DataRangeEntryForm.tsx

```diff
@@ -1,21 +1,42 @@
 import { DatePickerInput } from 'components/FormInputs/DatePickerInput';
 import { Controller, useFormContext, useWatch } from 'react-hook-form';
 import { FilterEntry } from './FilterEntry';
+import { isAfter } from 'date-fns';
+import { internalizeDate } from 'date';
+import { useEffect } from 'react';
 
 const DataRangeEntryForm = () => {
-    const { control } = useFormContext<FilterEntry, Partial<FilterEntry>>();
+    const { control, trigger } = useFormContext<FilterEntry, Partial<FilterEntry>>();
 
     const before = useWatch({ control, name: 'before' });
     const after = useWatch({ control, name: 'after' });
 
+    useEffect(() => {
+        trigger('after');
+    }, [before]);
+
+    const isBeforeTo =
```

Comment:
praise: Nice!
suggestion:  could this be made re-usable and moved into the [validation](https://github.com/CDCgov/NEDSS-Modernization/tree/main/apps/modernization-ui/src/validation) folder?  Might be able to start a folder for exclusively for date validations.